pkg/providers/instance/suite_test.go

```diff
@@ -147,6 +200,95 @@ var _ = Describe("VMInstanceProvider", func() {
 		Entry("non-zonal", azureEnvNonZonal, cloudProviderNonZonal),
 	}
 
+	Context("metrics integration", func() {
+		BeforeEach(func() {
+			instancemetrics.VMCreateStartMetric.Reset()
+			instancemetrics.VMCreateFailureMetric.Reset()
+		})
+
+		It("records VM create start metric during successful launch", func() {
+			ExpectApplied(ctx, env.Client, nodeClaim, nodePool, nodeClass)
+			pod := coretest.UnschedulablePod(coretest.PodOptions{})
+			ExpectProvisioned(ctx, env.Client, cluster, cloudProvider, coreProvisioner, pod)
+			ExpectScheduled(ctx, env.Client, pod)
+
+			Expect(azureEnv.VirtualMachinesAPI.VirtualMachineCreateOrUpdateBehavior.CalledWithInput.Len()).To(BeNumerically(">=", 1))
+			createInput := azureEnv.VirtualMachinesAPI.VirtualMachineCreateOrUpdateBehavior.CalledWithInput.Pop()
+			labels := vmMetricLabelsFromCreateInput(createInput, nodePool.Name)
+
+			metric, ok, err := metrics.FindMetricWithLabelValues("karpenter_instance_vm_create_start_total", labels)
```

Comment:
Just confirming that this asserts the labels + values match right? (99% sure it does)